### DISCUSSION TITLE
I used to have a 'Re-run all jobs' button in my GitHub runner, but it seems to have disappeared. I would appreciate any help in finding out how to get this button back.

### DISCUSSION THREAD

**Discussion by thapaashesh** [community/community#51503] [Actions] [answered]

**Comment by Anmol-Baranwal** [comment]
> It is recommended to write the main issue in body not title

GitHub recently made some changes to the Actions UI, which removed the 'Re-run all jobs' button from the runner page. Instead, you can now re-run all jobs from the Actions tab in your repository.

Here's how to re-run all jobs from the Actions tab:

1. Go to your repository on GitHub.
2. Click on the "Actions" tab.
3. Select the workflow run you want to re-run all jobs for.
4. Click on the three dots in the top-right corner of the workflow run and select "Re-run all jobs" from the dropdown menu.

This will trigger a new workflow run with all jobs re-run.

Alternatively, if you want to re-run a specific job, you can do so by following these steps:

1. Go to the workflow run you want to re-run the job for.
2. Click on the job you want to re-run.
3. Click on the "Re-run job" button in the top-right corner of the job log.

This will trigger a new workflow run with only the selected job re-run.

Feel free to ask, if you have any further doubts.

**Reply by JakeQZ** [reply]
> GitHub recently made some changes to the Actions UI, which removed the 'Re-run all jobs' button from the runner page. Instead, you can now re-run all jobs from the Actions tab in your repository.
> 
> Here's how to re-run all jobs from the Actions tab:
> 
>     1. Go to your repository on GitHub.
> 
>     2. Click on the "Actions" tab.
> 
>     3. Select the workflow run you want to re-run all jobs for.
> 
>     4. Click on the three dots in the top-right corner of the workflow run and select "Re-run all jobs" from the dropdown menu.

It doesn't work.  The only options on clicking the '...' next to the failed run are
- View #{PR numnber};
- View workflow file (which just opens the relevant `.yml` file);
- Delete workflow run.

You have to navigate to the specific workflow run by clicking on the title of the associated PR in the list of all workflow runs (the "Actions" tab).  When viewing the specific workflow run, a button to re-run jobs is next to the "..." near the top-right.

**Comment by thapaashesh** [comment]
appreciate your reply but i dont see "Re-run all jobs" when i click on  the three dots in the top-right corner from the dropdown menu.

**Reply by JakeQZ** [reply]
I don't either, and have replied to that effect.

**Reply by JakeQZ** [reply]
You have to navigate to the specific workflow run first by clicking on the title of the associated PR in the list of all workflow runs.

**Comment by Anmol-Baranwal** [comment, marked as answer]
> My apologies, it appears that our conversation got buried in the midst of other discussions.

![image](https://user-images.githubusercontent.com/74038190/231189435-c3290a99-3171-4d0b-b812-89204d6b8eb6.png)

The "Re-run all jobs" option is only available in the Actions tab of a GitHub repository, and only if there are previous workflow runs that have failed or been cancelled.

Here are the steps to find the "Re-run all jobs" option in the Actions tab of a GitHub repository:

1. Go to the repository's main page on GitHub.
2. Click on the "Actions" tab at the top of the page.
3. Find the failed or cancelled workflow run that you want to re-run, and click on it to open the workflow run page.
4. On the workflow run page, click on the "Re-run all jobs" button in the top-right corner of the page, next to the "Cancel workflow" button.

If there are no previous workflow runs that have failed or been cancelled, the "Re-run all jobs" option will not be available. In that case, you can manually trigger a new workflow run by pushing a new commit or by using the "Run workflow" button in the Actions tab.

**Reply by Anmol-Baranwal** [reply]
@thapaashesh 
I just wanted to check in and see if you have any remaining questions or concerns about my answer. If not, would you please mark my response as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by thapaashesh** [reply]
@Anmol-Baranwal Please accept my sincere apologies on delayed response. Thank you so much for your assistance.

**Reply by Anmol-Baranwal** [reply]
No worries at all! I'm here to help whenever you need it :)

**Reply by JakeQZ** [reply]
OK.  That works.  But it's ridiculously cumbersome.  Previously there was a 're-run' button that was much more convenient.

The workflows often fail due to server errors.  M$ are obviously trying to limit access because their servers can't handle the demand.  One way of doing that is to obfuscate access to re-runs.  But that is counter-productive, and it should be obvious why.

**Comment by nhra168** [comment]
Really support

**Comment by TomSweeneyRedHat** [comment]
I'm trying actins for the first time and have a new workflow that I'm trying to get to work.  It's failing and has never run.  I can rerun it by changing a file the workflow is watching, but when I go to the Actions tab, there is no "re-run" option under the three dots.  Only "Create status badge", "Pin workflow", and "Disable workflow".  

It would be nice to also have a re-run button, but I can trigger the action in other ways.

**Reply by JakeQZ** [reply]
I found that from a PR where it says 'Some checks were not successful' (or 'All checks have passed'), you can click on the "..." button next to any action (failing or successful), then click "View details".  On the details page for the specific action, there's the "Re-run all jobs" button (or drop-down with "Re-run failing jobs" option also available if some failed) buried top-right of that page.  HTH.

**Comment by lingsamuel** [comment]
Same here. The button was there at least a week ago.
<img width="1533" height="474" alt="image" src="https://github.com/user-attachments/assets/02c16a6e-7251-4c85-a744-1f8c8be0a614" />

**Comment by juanluisbaptiste** [comment]
Same here, until a couple months ago I could re-run any workflow, no matter it was failed or successful. 

For example, this one that succeeded I normally re-run it to update the docker base image when Snyk reports security vulnerabilities, without the need to do a new release, now the re-run workflow option is gone: 

<img width="436" height="232" alt="Screenshot_20251211_125348" src="https://github.com/user-attachments/assets/1450f964-12a4-44f1-bac2-d2eb283cbb5e" />

**Reply by JakeQZ** [reply]
If you're using a phone, the option is in the `...` menu.

**Reply by juanluisbaptiste** [reply]
I'm not using a phone, and the re-run option also isn't in the ... menu
<img width="285" height="172" alt="Screenshot_20251211_131230" src="https://github.com/user-attachments/assets/124c2adc-ecba-4444-888d-89962c1c579a" />

**Reply by JakeQZ** [reply]
I see this on what looks like the equivalent page, so I don't know why it isn't working for you:

<img width="310" height="68" alt="image" src="https://github.com/user-attachments/assets/eafe2696-9959-44d7-9176-e84196c6c778" />

Could it be a browser cache issue (e.g. some JavaScript that sets up the buttons has not been updated)?

**Reply by juanluisbaptiste** [reply]
I tested that too, deleting cookies and logging again, no change.

**Reply by johntruckenbrodt** [reply]
The button disappears after 30 days:
https://docs.github.com/en/actions/how-tos/manage-workflow-runs/re-run-workflows-and-jobs#about-re-running-workflows-and-jobs
